anvill/src/Lifters/FunctionLifter.cpp

```diff
@@ -317,10 +327,24 @@ void FunctionLifter::VisitConditionalFunctionReturn(
   MuteStateEscape(
       remill::AddTerminatingTailCall(taken_block, intrinsics.function_return));
   VisitDelayedInstruction(inst, delayed_inst, not_taken_block, false);
-  llvm::BranchInst::Create(GetOrCreateBlock(inst.branch_not_taken_pc),
+  llvm::BranchInst::Create(GetOrCreateTargetBlock(inst.branch_not_taken_pc),
                            not_taken_block);
 }
 
+std::optional<FunctionDecl> FunctionLifter::TryGetTargetFunctionType(std::uint64_t address) {
+  auto redirected_addr = control_flow_provider->GetRedirection(address);
+
+  auto opt_function_decl = type_provider.TryGetFunctionType(redirected_addr);
+  if (!opt_function_decl.has_value()) {
+    return std::nullopt;
```

Comment:
Don't fail here, fall through to try to get the `function_decl` for the original address.

Reply:
So the logic should be:

```
if (!opt_function_decl.has_value()) {
  if (address != redirected_address) {
    opt_function_decl = type_provider.TryGetFunctionType(address);
    if (opt_function_decl.has_value()) {
      opt_function_decl->address = redirected_addr;
    }
  }
}
return opt_function_decl;
```